### PR TITLE
Tidy up usage of channels for signaling

### PR DESCRIPTION
**Pull request by marquiz**
This started as a small effort to simplify the usage of "ready" channel in nfd-master. It extended into a wider simplification/unification of the channel usage.

### CODE REVIEW

File: pkg/nfd-master/nfd-master.go [hunk 1]
```diff
@@ -475,11 +474,8 @@ func (m *nfdMaster) Stop() {
 // Wait until NfdMaster is able able to accept connections.
 func (m *nfdMaster) WaitForReady(timeout time.Duration) bool {
 	select {
-	case ready, ok := <-m.ready:
-		// Ready if the flag is true or the channel has been closed
-		if ready || !ok {
-			return true
-		}
+	case <-m.ready:
+		return true
```
Comment (ArangoGutierrez) [inline pkg/nfd-master/nfd-master.go:478]: I think we need to re write the entire func, the last return will never be used

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

File: pkg/nfd-master/nfd-master.go (hunk 1 above)
Comment (marquiz) [reply pkg/nfd-master/nfd-master.go:478]: > I think we need to re write the entire func, the last return will never be used

True, I modified the func to eliminate unreachable parts

Comment (ArangoGutierrez) [reply pkg/nfd-master/nfd-master.go:478]: pardon my ignorance, should this `select` be inside an empty `for` ?

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

File: pkg/nfd-master/nfd-master.go (hunk 1 above)
Comment (marquiz) [reply pkg/nfd-master/nfd-master.go:478]: No, the select blocks until the channel is closed or the timer expoires

**Pull request merged**